### PR TITLE
feat: add initial support for super reactions

### PR DESCRIPTION
**Pull request by Mateo-tem**
**Please describe the changes this PR makes and why it should be merged:**

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**

- https://github.com/discord/discord-api-docs/pull/6056

### CODE REVIEW

File: deno/payloads/v10/channel.ts [hunk 1]
```diff
@@ -856,16 +856,42 @@ export interface APIReaction {
...
+	/**
+	 * HEX colors used for super reaction
+	 */
+	burst_colors: string[];
+}
+
+/**
+ * https://discord.com/developers/docs/resources/channel#reaction-count-details-object-reaction-count-details-structure
+ */
+export interface ReactionCountDetails {
```
Comment (vladfrangu) [inline deno/payloads/v10/channel.ts:886]: Prefix with API

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — DISMISSED** [review]

**Comment by JJRcop** [comment]
MESSAGE_REACTION_ADD also has updates, which aren't in this PR. Is this usable yet or still subject to changes?

<details>
<summary>Screenshots</summary>

Screenshot_20230520_103159

Screenshot_20230520_102004
</details>

**Comment by Syjalo** [comment]
We only add what is documented (or will be documented). Suggest changes in the docs PR.
- https://github.com/discord/discord-api-docs/pull/6056

**Comment by JJRcop** [comment]
Ah, I misplaced the repo this was on, thanks.

**Review by almeidx — CHANGES_REQUESTED** [review]
The `APIReaction` interface is missing the `me_burst` property, and some of the descriptions are outdated

File: deno/rest/v10/channel.ts [hunk 2]
```diff
@@ -337,6 +337,10 @@ export type RESTDeleteAPIChannelMessageUserReactionResult = never;
  * https://discord.com/developers/docs/resources/channel#get-reactions
  */
 export interface RESTGetAPIChannelMessageReactionUsersQuery {
+	/**
+	 * The reaction type (0 for normal and 1 for super)
+	 */
+	type: number;
```
Comment (vladfrangu) [inline deno/rest/v10/channel.ts:343]: Should be turned into an enum

**Review by vladfrangu — DISMISSED** [review]

**Comment by Jiralite** [comment]
Hey @Mateo-tem, are you able to split this pull request up so we have something to merge?

**Comment by Mateo-tem** [comment]
> Hey @Mateo-tem, are you able to split this pull request up so we have something to merge?

Yes! I'll do that in a moment 🫡

File: deno/payloads/v10/channel.ts [hunk 3]
```diff
@@ -900,19 +900,45 @@ export interface APIFollowedChannel {
...
+	 */
+	me_burst: boolean;
 	/**
 	 * Emoji information
 	 *
 	 * See https://discord.com/developers/docs/resources/emoji#emoji-object
 	 */
 	emoji: APIPartialEmoji;
+	/**
+	 * HEX colors used for super reaction
```
Comment (Jiralite) [inline deno/payloads/v10/channel.ts:925]:
```suggestion
	 * Hexadecimal colors used for this super reaction
```

File: deno/payloads/v9/channel.ts [hunk 4]
```diff
@@ -865,19 +865,45 @@ export interface APIFollowedChannel {
...
+	 */
+	me_burst: boolean;
 	/**
 	 * Emoji information
 	 *
 	 * See https://discord.com/developers/docs/resources/emoji#emoji-object
 	 */
 	emoji: APIPartialEmoji;
+	/**
+	 * HEX colors used for super reaction
```
Comment (Jiralite) [inline deno/payloads/v9/channel.ts:890]:
```suggestion
	 * Hexadecimal colors used for this super reaction
```

File: payloads/v10/channel.ts [hunk 5]
```diff
@@ -900,19 +900,45 @@ export interface APIFollowedChannel {
...
+	 */
+	me_burst: boolean;
 	/**
 	 * Emoji information
 	 *
 	 * See https://discord.com/developers/docs/resources/emoji#emoji-object
 	 */
 	emoji: APIPartialEmoji;
+	/**
+	 * HEX colors used for super reaction
```
Comment (Jiralite) [inline payloads/v10/channel.ts:925]:
```suggestion
	 * Hexadecimal colors used for this super reaction
```

File: payloads/v9/channel.ts [hunk 6]
```diff
@@ -865,19 +865,45 @@ export interface APIFollowedChannel {
...
+	 */
+	me_burst: boolean;
 	/**
 	 * Emoji information
 	 *
 	 * See https://discord.com/developers/docs/resources/emoji#emoji-object
 	 */
 	emoji: APIPartialEmoji;
+	/**
+	 * HEX colors used for super reaction
```
Comment (Jiralite) [inline payloads/v9/channel.ts:890]:
```suggestion
	 * Hexadecimal colors used for this super reaction
```

File: deno/payloads/v10/channel.ts [hunk 7]
```diff
@@ -900,19 +900,45 @@ export interface APIFollowedChannel {
  */
 export interface APIReaction {
 	/**
-	 * Times this emoji has been used to react
+	 * Total number of times this emoji has been used to react (including super reacts)
 	 */
 	count: number;
+	/**
+	 * Reaction count details object
```
Comment (Jiralite) [inline deno/payloads/v10/channel.ts:907]: This description just echoes the name of the interface, like it's as useful as not including it. Let's improve it:

```suggestion
	 * An object detailing the individual reaction counts for different types of reactions 
```

File: deno/payloads/v9/channel.ts [hunk 8]
```diff
@@ -865,19 +865,45 @@ export interface APIFollowedChannel {
  */
 export interface APIReaction {
 	/**
-	 * Times this emoji has been used to react
+	 * Total number of times this emoji has been used to react (including super reacts)
 	 */
 	count: number;
+	/**
+	 * Reaction count details object
```
Comment (Jiralite) [inline deno/payloads/v9/channel.ts:872]: This description just echoes the name of the interface, like it's as useful as not including it. Let's improve it:

```suggestion
	 * An object detailing the individual reaction counts for different types of reactions 
```

File: payloads/v10/channel.ts [hunk 9]
```diff
@@ -900,19 +900,45 @@ export interface APIFollowedChannel {
  */
 export interface APIReaction {
 	/**
-	 * Times this emoji has been used to react
+	 * Total number of times this emoji has been used to react (including super reacts)
 	 */
 	count: number;
+	/**
+	 * Reaction count details object
```
Comment (Jiralite) [inline payloads/v10/channel.ts:907]: This description just echoes the name of the interface, like it's as useful as not including it. Let's improve it:

```suggestion
	 * An object detailing the individual reaction counts for different types of reactions 
```

File: payloads/v9/channel.ts [hunk 10]
```diff
@@ -865,19 +865,45 @@ export interface APIFollowedChannel {
  */
 export interface APIReaction {
 	/**
-	 * Times this emoji has been used to react
+	 * Total number of times this emoji has been used to react (including super reacts)
 	 */
 	count: number;
+	/**
+	 * Reaction count details object
```
Comment (Jiralite) [inline payloads/v9/channel.ts:872]: This description just echoes the name of the interface, like it's as useful as not including it. Let's improve it:

```suggestion
	 * An object detailing the individual reaction counts for different types of reactions 
```

**Review by Jiralite — CHANGES_REQUESTED** [review]

**Review by almeidx — APPROVED** [review]

**Pull request merged**